packages/vite/src/esbuild-request.ts

```diff
@@ -40,14 +52,24 @@ export class EsBuildModuleRequest implements ModuleRequest {
     readonly isVirtual: boolean,
     readonly isNotFound: boolean,
     readonly resolvedTo: Resolution<OnResolveResult, OnResolveResult> | undefined
-  ) {}
+  ) {
+    let plugins = (this.context.initialOptions.plugins ?? []).map(p => p.name);
+    if (plugins.includes('vite:dep-pre-bundle')) {
+      this.phase = 'bundling';
+    } else if (plugins.includes('vite:dep-scan')) {
+      this.phase = 'scanning';
+    } else {
+      throw new Error(`cannot identify what phase vite is in. Saw plugins: ${plugins.join(', ')}`);
```

Comment:
This phase detection is extremely tied to vite internals. But we need to distinguish scanning from bundling because scanning already has the entire vite resolver configured whereas bundling is much more primitive and needs us to do more.